exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetScanBatchCreator.java

```diff
@@ -87,9 +87,6 @@ public ScanBatch getBatch(FragmentContext context, ParquetRowGroupScan rowGroupS
           newColumns.add(column);
         }
       }
-      if (newColumns.isEmpty()) {
```

Comment:
I went through all the ScanBatchCreator in Drill's code base. Seems ParquetScanBatchCreator is the only one that is converting an empty column list to ALL_COLUMNS. Looking at the history, seems DRILL-1845 added the code, probably just to make it work in parquet for skipAll query.  

With the patch of DRILL-4279, parquet record reader would be able to handle empty column list. 

Besides ParquetScanBatchCreator, this patch also modifies HBaseGroupScan, EasyGroupScan where it originally interprets empty column lists into ALL_COLUMNS. 

I'll add some comment to the code to clarify the different meaning of NULL and empty column list.